### PR TITLE
declare list types

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/nikic/PHP-Parser/issues/905

### CODE REVIEW

**Comment by nikic** [comment]
Yeah, that was indeed a bug. I've applied the fix separately in https://github.com/nikic/PHP-Parser/commit/21a3e8cac52b1d3d0ea0e91c34d112cf53716732 and cherry-picked this into 4.x as well. Can you please rebase?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -85,3 +85,25 @@ jobs:
...
+        uses: "actions/checkout@v2"
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "none"
+          php-version: "${{ matrix.php-version }}"
+          tools: composer:v2
+      - name: "Install dependencies"
+        run: | 
+          composer global require phpstan/phpstan:1.9.3
```
Comment (nikic) [inline .github/workflows/main.yml:106]: Can we install this from `tools/composer.json` instead?

File: lib/PhpParser/Node/Attribute.php [hunk 2]
```diff
@@ -9,12 +9,12 @@ class Attribute extends NodeAbstract {
...
     public $name;
 
-    /** @var Arg[] Attribute arguments */
+    /** @var list<Arg> Attribute arguments */
     public $args;
 
     /**
      * @param Node\Name $name       Attribute name
-     * @param Arg[]     $args       Attribute arguments
+     * @param list<Arg>     $args       Attribute arguments
```
Comment (nikic) [inline lib/PhpParser/Node/Attribute.php:17]:
```suggestion
     * @param list<Arg> $args Attribute arguments
```

File: lib/PhpParser/Node/Stmt/EnumCase.php [hunk 3]
```diff
@@ -16,7 +16,7 @@ class EnumCase extends Node\Stmt {
     /**
      * @param string|Node\Identifier    $name       Enum case name
      * @param Node\Expr|null            $expr       Enum case expression
-     * @param AttributeGroup[]          $attrGroups PHP attribute groups
+     * @param list<AttributeGroup>          $attrGroups PHP attribute groups
```
Comment (nikic) [inline lib/PhpParser/Node/Stmt/EnumCase.php:19]:
```suggestion
     * @param list<AttributeGroup> $attrGroups PHP attribute groups
```

File: lib/PhpParser/Node/Param.php [hunk 4]
```diff
@@ -31,7 +31,7 @@ class Param extends NodeAbstract {
      * @param bool                                    $variadic   Whether this is a variadic argument
      * @param array<string, mixed> $attributes Additional attributes
      * @param int                                     $flags      Optional visibility flags
-     * @param AttributeGroup[]                        $attrGroups PHP attribute groups
+     * @param list<AttributeGroup>                        $attrGroups PHP attribute groups
```
Comment (nikic) [inline lib/PhpParser/Node/Param.php:34]:
```suggestion
     * @param list<AttributeGroup> $attrGroups PHP attribute groups
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (staabm) [reply .github/workflows/main.yml]: done.

**Comment by staabm** [comment]
should be good to go, thx

**Pull request merged**